### PR TITLE
feature: implement contest.list module

### PR DESCRIPTION
**Pull request by MohanadKh03**

### CODE REVIEW

File: src/modules/contest/list/list.test.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    const client = new CodeforcesClient(process.env.CF_KEY, process.env.CF_SECRET)
+    const result = await client.contest.list({gym: true});
+    
+    expect(result.status).toBe('OK');
+
+    if (result.status === 'OK') {
+        const contestByAcmASCIS = result.result.some(contest => contest.id === acmascisPrivateContestId);
+        expect(contestByAcmASCIS).toBe(true);
+    }
+  },ONE_MINUTE_MS);
```
Comment (khaled-hamam) [inline src/modules/contest/list/list.test.ts:17]: Question: We didn't need this timeout for the other tests? Any clue if they're still working or not?

File: src/modules/contest/list/list.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+type ContestListParams = {
+    /* 
+     * When it is null then codeforces treats it as false by default only public contests(not mashups/gums) are returned
+    */
+    gym?: boolean; 
+};
+
+
+export const list = (client: ICodeforcesClient) => async (params: ContestListParams) => {
```
Comment (khaled-hamam) [inline src/modules/contest/list/list.ts:12]: [Nitpick] I guess we can start a good practice by leaving a link to the original endpoint documentation in the code.

https://codeforces.com/apiHelp/methods#contest.list

**Review by khaled-hamam — APPROVED** [review]
Just left a question and a nitpick. Feel free to skip the nitpick for now.

File: src/modules/contest/list/list.test.ts (hunk 1 above)
Comment (MohanadKh03) [reply src/modules/contest/list/list.test.ts:17]: Unlike other methods which take about 5 records ``count : 5`` , this one actually gets ALL the contests and by using ``gym : true`` it is listing all contests and gyms available and sadly there is no count or any other parameter that we can use to filter or get less contests 
Each test takes about 15 seconds ish 
And Jest's default timeout per test is 5 seconds https://jestjs.io/docs/api#testname-fn-timeout

**Pull request merged**